code/clue.js

```diff
@@ -1,59 +1,350 @@
+};
+
+const favWeapons = [rope, knife, candlestick, dumbbell, poison, axe];
+const shuffleFavouriteWeapon = () => {
+  mystery.favWeapon = shuffle(favWeapons, weapons);
+  document.getElementById(
+    "mysteryName"
+  ).innerHTML = `Favorite wepon is ${mystery.favWeapon.id}`;
+};
+
+const shuffle = (favArr, arr) => {
+  const favDuplicated = duplicateItems(favArr);
+  const concatArr = [...arr, ...favDuplicated];
+  return randomSelector(concatArr);
+};
+// ----------------WEAPON AND ROOM
+
+const shuffleFavRoom = () => {
+  if (mystery.weapon !== undefined) {
+    const roomList = weaponRooms[mystery.weapon.id];
+    mystery.room = shuffle(roomList, rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  } else {
+    mystery.room = randomSelector(rooms);
+    document.getElementById("roomName").innerHTML = mystery.room;
+  }
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
   // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
 
   // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
+
+  document.getElementById("killerCard").style.background = mystery.killer.color;
+
+  document.getElementById("killerName").innerHTML =
+    mystery.killer.firstName + " " + mystery.killer.lastName;
+
+  document.getElementById("killerOccupation").innerHTML =
+    mystery.killer.occupation;
+
   document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+    "killerImage"
+  ).innerHTML = `<img src="${mystery.killer.image}">`;
+};
 
 // CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
 
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons);
+  document.getElementById(
+    "weaponName"
+  ).innerHTML = `${mystery.weapon.name} which weight is ${mystery.weapon.weight}.`;
+};
+
+// LOADER
+
+document.getElementById("loaderItem").style.display = "none";
+
+// Invoke function
+
+document.getElementById("killerCard").onclick = () => {
+  loader();
+  pickKiller();
+};
+
+document.getElementById("weaponCard").onclick = () => {
+  loader();
+  pickWeapon();
+};
+document.getElementById("roomCard").onclick = () => {
+  loader();
+  // pickRoom();
+  shuffleFavRoom();
+};
+
+document.getElementById("mysteryCard").onclick = shuffleFavouriteWeapon;
+
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+// 'The murder was committed by Jacob Green, in the living room with a rope. '
+
+const revalMystery = () => {
```

Comment:
Just as an idea: It could be nice to include a message to the user if not all of the four decks have been clicked, something like "Please choose one card from each deck". Otherwise, the user could start to wonder if the "Reveal the mystery" button is working properly.

Reply:
Updated